NitroxInstallerActions/Patches/NitroxServerSetup.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace InstallerActions.Patches
+{
+    public static class NitroxServerSetup
+    {
+        private static string assemblyCSharpDLL = "Assembly-CSharp.dll";
+        private static string serverAssemblyCSharpDLL = "..\\..\\SubServer\\" + assemblyCSharpDLL;
+
+        private static string assemblyCSharpfirstpassDLL = "Assembly-CSharp-firstpass.dll";
+        private static string serverAssemblyCSharpfirstpassDLL = "..\\..\\SubServer\\" + assemblyCSharpfirstpassDLL;
+
+        private static string newtonsoftJsonDLL = "Newtonsoft.Json.dll";
+        private static string serverNewtonsoftJsonDLL = "..\\..\\SubServer\\" + newtonsoftJsonDLL;
+
+        private static string unityEngineDLL = "UnityEngine.dll";
+        private static string serverUnityEngineDLL = "..\\..\\SubServer\\" + unityEngineDLL;
+
+        private static string unityEngineUIDLL = "UnityEngine.UI.dll";
+        private static string serverunityEngineUIDLL = "..\\..\\SubServer\\" + unityEngineUIDLL;
+
+        private static string poly2TriDLL = "Poly2Tri.dll";
+        private static string serverPoly2TriDLL = "..\\..\\SubServer\\" + poly2TriDLL;
+
+        private static string litJsonDLL = "LitJson.dll";
+        private static string serverLitJsonpDLL = "..\\..\\SubServer\\" + litJsonDLL;
+
+        private static string iTweenDLL = "iTween.dll";
+        private static string serverITweenDLL = "..\\..\\SubServer\\" + iTweenDLL;
+
+        private static string libNoiseDLL = "LibNoise.dll";
+        private static string serverLibNoiseDLL = "..\\..\\SubServer\\" + libNoiseDLL;
+
+        private static string lumenWorksDLL = "LumenWorks.dll";
+        private static string serverLumenWorksDLL = "..\\..\\SubServer\\" + lumenWorksDLL;
+
+        // dir = (Subnautica Directory)\Subnautica_Data\Managed
+        public static void Init(string dir)
+        {
+            string file = Path.Combine(Path.GetFullPath(dir), serverAssemblyCSharpDLL); // (Subnautica Directory)\SubServer\Assembly-CSharp.dll
+            if (File.Exists(file))
+            {
+                File.Delete(file);
+            }
+            File.Copy(dir + assemblyCSharpDLL, file);
```

Comment:
You can refactor all of these out into a single method:

```
private void copyAssembly(String assemblyName, String directory) {
             string file = Path.Combine(Path.GetFullPath(dir), assemblyName); 
             if (File.Exists(file))
             {
                 File.Delete(file);
             }
             File.Copy(dir + assemblyName, file);
}
````

then do 

```
foreach(String assembly in assemblies) {
    String assemblyPath = "..\\..\\SubServer\\" + assembly;
    copyAssembly(path, dir);
}
```

Reply:
the full path is already passed through the dir variable btw